[WebSocket] Add configuration metadata store url setting in websocket

RobertIndie opened the pull request:
Master Issue: #13760 

### Motivation

As per #13077, `configurationStoreServers` need to be renamed to `configurationMetadataStoreUrl`.

### Modifications

* Add `configurationMetadataStoreUrl` setting to websocket
* Deprecate `configurationStoreServers` in websocket

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

- [x] `doc-required` 
  
#13758 contains doc changes for this PR.

codelipenghui commented on pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:
```diff
@@ -67,9 +67,13 @@
     private String configurationStoreServers;
 
+    @FieldContext(doc = "Connection string of configuration metadata store servers")
+    private String configurationMetadataStoreUrl;
```
Please help also update the `websocket.conf`, following the same pattern of `broker.conf` to move the deprecated configurations to the end of the config file.

Anonymitaet commented:
Hi @RobertIndie I've removed `doc-required` and added `doc-added` since the doc has been added.

BTW, https://github.com/apache/pulsar/pull/13758 can be merged after this PR gets merged?

RobertIndie commented:
> Hi @RobertIndie I've removed `doc-required` and added `doc-added` since the doc has been added. Please pay attention to this next time, thanks!

Thanks.

> BTW, #13758 can be merged after this PR gets merged?

Yes. #13758 already contains the doc for this PR.

RobertIndie commented:
/pulsarbot run-failure-checks

nodece commented on pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:
```diff
@@ -67,9 +67,13 @@
     private String configurationStoreServers;
 
+    @FieldContext(doc = "Connection string of configuration metadata store servers")
+    private String configurationMetadataStoreUrl;
```
Looks like we should be compatible with the old config, overwrite the `getConfigurationMetadataStoreUrl()`, so like:
```java
    public String getConfigurationMetadataStoreUrl() {
        if (StringUtils.isNotEmpty(this.configurationMetadataStoreUrl)) {
            return this.configurationMetadataStoreUrl;
        }

        // backward compatible
        if (StringUtils.isNotEmpty(this.configurationStoreServers)) {
            return this.configurationStoreServers;
        }
        return this.globalZookeeperServers;
    }
```

RobertIndie replied:
This has already been done in `ServiceConfiguration.getConfigurationMetadataStoreUrl`. When initializing the `WebSocketService`, we will convert `WebSocketProxyConfiguration` to `ServiceConfiguration` via `PulsarConfigurationLoader.convertFrom`. Backward compatibility has been done in `ServiceConfiguration`. 

> You also need to check the usage, use `WebSocketProxyConfiguration#getConfigurationMetadataStoreUrl()` instead of `globalZookeeperServers()` and `getConfigurationStoreServers()`.

I have already checked that. There are no direct use of `WebSocketProxyConfiguration#getConfigurationMetadataStoreUrl()` and there shouldn't be.

nodece reviewed (approved):
LGTM
